micro_epsilon_scancontrol_driver/src/driver.cpp

```diff
@@ -550,6 +564,100 @@ namespace scancontrol_driver
         // return true;
     }  
 
+    // Generic function for setting a times like exposure and idle time.
+    // setting_id: 
+    // value: time in microseconds. min: 1 mus, max: 40950 mus ;  eg. 1005 = 1.005ms
+    int ScanControlDriver::SetTime(unsigned int setting_id, unsigned int value){
+        int ret_code;
+
+        // encoded value in 1 mus steps. 
+        uint32_t remainder = ((value % 10) << 12) & 0xF000; // Remainder is left shifted first and bits 0-12 are masked. the quotient will occupy this area.
+        uint32_t quotient = ((value / 10)) & 0xFFF; // take the quotient and mask bits 12-15
+        uint32_t value_encoded = remainder + quotient; 
+
+        ret_code = SetFeature(setting_id, value_encoded); 
+        
+        // success if value was sent AND set.
+        if (ret_code < GENERAL_FUNCTION_OK){
+            RCLCPP_INFO_STREAM(LOGGER, "SetFeature failed.");
```

Comment:
- Why did it fail, if unknown mention it and also print the return_code for somewhat easier debugging? 
- As you return the result code, is the function calling this function not responsible for the logging?
- Should this not be a error serverity?

Reply:
1. Added return code in string.
2. The return code is still needed as an output to fill the service message. Otherwise it's not necessary. I could keep it as an argument that gets filled by reference if that's more appropriate.
3. Yes. changed.